behaviour/graql/language/undefine.feature

```diff
@@ -29,215 +29,1173 @@ Feature: Graql Undefine Query
+      | EMA |
+      | ATT |
+    Given graql insert
       """
-      undefine person key email;
+      insert $x "Colette" isa name;
+      """
+    Given the integrity is validated
+    Then graql undefine throws
+      """
+      undefine name sub attribute;
+      """
+    Then the integrity is validated
+    When graql delete
+      """
+      match
+        $x isa name;
+      delete
+        $x isa name;
+      """
+    Then the integrity is validated
+    When graql undefine
+      """
+      undefine name sub attribute;
       """
     Then the integrity is validated
     When get answers of graql query
       """
-      match $x type child; $x key email; get;
+      match $x sub attribute; get;
       """
-    Then answer size is: 0
+    Then uniquely identify answer concepts
+      | x   |
+      | EMA |
+      | ATT |
 
 
-  @ignore
-  # re-enable when 'relates' is inherited
-  Scenario: undefine 'relates' from super relation removes 'relates' from child relation
-    Given graql define
+  ########################
+  # ATTRIBUTE OWNERSHIPS #
+  ########################
+
+  Scenario: undefining an attribute ownership removes it
+    Given get answers of graql query
       """
-      define part-time sub employment; 
+      match
+        $x has name;
+        $x type person;
+      get;
       """
-    Given graql undefine
+    Given concept identifiers are
+      |     | check | value  |
+      | PER | label | person |
+    Given uniquely identify answer concepts
+      | x   |
+      | PER |
+    When graql undefine
       """
-      undefine employment relates employer;
+      undefine person has name;
       """
     Then the integrity is validated
     When get answers of graql query
       """
-      match $x type part-time; $x relates $role; get; 
+      match
+        $x has name;
+        $x type person;
+      get;
       """
-    Then concept identifiers are
-      |           | check | value     |
-      | EMPLOYEE  | label | employee  |
-      | PART_TIME | label | part-time |
-    Then uniquely identify answer concepts
-      | x         | role     |
-      | PART_TIME | EMPLOYEE |
+    Then answer size is: 0
 
-  @ignore
-  # re-enable when 'relates' is bound to a relation and blockable
-  # TODO
-  Scenario: undefine 'relates' from super relation that is overriden using 'as' removes override from child (?)
 
-  @ignore
-  # TODO
-  Scenario: undefine a sub-role using 'as' removes sub-role from child relations
+  Scenario: attempting to undefine an attribute ownership that was not actually owned to begin with throws an error
+    Then graql undefine throws
+      """
+      undefine employment has name;
+      """
+    Then the integrity is validated
 
-  Scenario: undefine 'plays' from super relation removes 'plays' from child relation
 
-  Scenario: undefine 'has' from super relation removes 'has' from child relation
+  Scenario: attempting to undefine an attribute ownership inherited from a parent throws an error
```

Comment:
and i believe we decided this would be A.O.K

Reply:
no